doc/source/dev/starting_galaxy.md

```diff
@@ -0,0 +1,90 @@
+# Starting Galaxy
+
+## What happens when you start Galaxy?
+
+Galaxy is started by running ./run.sh script. The following call graph documents what happens during Galaxy startup process. 
+
+* ./galaxy/scripts/paster.py
+  * Add ./galaxy/lib to system path
+  * Check Python version
+    * version >= 3.5 -> OK
+    * version = 2.7 -> Deprecated
+    * No other version is supported
+  * serve.run()
+
+* ./galaxy/lib/galaxy/util/pastescript/serve.py (The Python Paste package contains Python modules that help in implementing WSGI middleware)
+  * serve.run()
+    * Parse Python command line arguments
+    * Extract the command name from command line arguments (e.g., 'serve')
+    * Get the Command object for command name, then invoke it (call Command's run() method)
+  * Command.run()
+    * Setup default options, validate command line arguments, then call ServeCommand's command() method
+  * ServeCommand.command()
+    * Conditionally, stop the daemon 
+    * If a config file is required, make sure a config file is provided on the command line arguments.
+    * Possible subcommands: start, stop, restart, status (or None)
+    * Call loadserver() method in ./galaxy/lib/galaxy/util/pastescript/loadwsgi.py
+      * Calls loadobj()  -> calls LoaderContext's create() -> calls _Server's invoke()  
+    * Call loadapp() method in ./galaxy/lib/galaxy/util/pastescript/loadwsgi.py
+      * Calls loadobj()  -> calls LoaderContext's create() -> calls _App's invoke() -> calls fix_call() on app_factory (in ./galaxy/lib/galaxy/webapps/galaxy/buildapp.py)
```

Comment:
This part is specific to Paste. This is also not the typical way Galaxy starts if you start it via `run.sh` which uses uwsgi by default. If you can just add a note about that I think that'd be sufficient. Galaxy is a WSGI app, so in principle all python WSGI servers can run the Galaxy app (paster, gunicorn and uwsgi are known to work and have some level of support).

Reply:
I added the note and pushed.

Reply:
Now it sounds like the entire startup is completely different. These paths all converge on instantiating a UniverseApplication instance. See for instance https://github.com/galaxyproject/galaxy/blob/278a0333724e84209f317bc200c86ad5165f575a/scripts/galaxy-main#L105 which is probably the easiest way to build a Galaxy app that is not serving the WSGI life-cycle.

Reply:
Should I revise the doc in anyway?

Reply:
Or can it be merged as is?